src/test/java/seedu/address/logic/commands/ViewGroupCommandTest.java

```diff
@@ -0,0 +1,124 @@
+//@@author hthjthtrh
+package seedu.address.logic.commands;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static seedu.address.commons.core.Messages.MESSAGE_EXECUTION_FAILURE;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.ViewGroupCommand.MESSAGE_GROUP_NONEXISTENT;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.UndoRedoStack;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+
+public class ViewGroupCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+    private ViewGroupCommand viewGrpCmd;
+
+    @Before
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+    }
+
+    /**
+     * creates the viewgroupcommand depending on grpName and idx
+     */
+    private ViewGroupCommand prepareCommand(String grpName, Index idx) {
+        if (grpName != null) {
+            viewGrpCmd = new ViewGroupCommand(grpName);
+        } else {
+            viewGrpCmd = new ViewGroupCommand(idx);
+        }
+        viewGrpCmd.setData(model, new CommandHistory(), new UndoRedoStack());
+        return viewGrpCmd;
+    }
+
+    /*
```

Comment:
Again, remember to remove unused code from the main codebase. If you want to include it still, include it under your own collated folder.